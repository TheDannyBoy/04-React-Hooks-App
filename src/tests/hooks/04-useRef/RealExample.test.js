import { shallow } from "enzyme";
import { RealExample } from "../../../components/04-useRef/RealExample";

describe('Testing RealExample', () => {
  test('should show correctly', () => {
    const wrapper = shallow(<RealExample />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
  });

  test('should show component <MultipleCustomHooks>', () => {
    const wrapper = shallow(<RealExample />);
    wrapper.find('button').simulate('click');
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  });
});
