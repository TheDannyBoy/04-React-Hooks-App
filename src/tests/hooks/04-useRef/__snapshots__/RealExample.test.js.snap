// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RealExample should show correctly 1`] = `
<div>
  <h1>
    Real Example
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
